js/autotrack/textInput.js

```diff
@@ -37,3 +42,57 @@ const debouncedAutocaptureTextInputChange = track => (
 
   track(eventType, autotrackProps);
 };
+
+// :HACK: In previous implementations, there would be a 'TextInput' component somewhere in the hierarchy.  However, with this
+// implementation, this is not the case (even the TextInputComponent has a displayName of 'InternalTextInput').  This means that any event
+// definitions on 'TextInput' would break.
+// To get around this, wrap the returned HOC with a no-op component with the 'TextInput' display name.
+// :TODO: (jmtaber129): Consider other workarounds to this, like setting the display name of instrumented components when they're exported
+// from the React Native lib (via instrumentation).
+const NoopTextInput = props => {
+  return props.children;
+};
+
+NoopTextInput.displayName = 'TextInput';
+
+export const withHeapTextInputAutocapture = track => TextInputComponent => {
+  class HeapTextInputAutocapture extends React.Component {
```

Comment:
wondering if this needs to be a class component since its not really stateful

Reply:
The `autocaptureTextInputChangeWithDebounce` is effectively state (not the traditional React `state`, but the sort of object state you'd want a class component for).